`alignas` on C23

In the latest MacOS CI jobs there's the following failure:
```
Starting test 229/455 on Mon Sep  1 12:13:06 2025: testc11/alignas
alignas.c[6:0-0] : syntax error
Parsing errorFatal error: exception GoblintCil__Frontc.ParseError("Parse error")
make: *** [testc11/alignas] Error 2
```
This seems to be because C23 replaces `alignas` macro with a keyword.

--- a/src/frontc/clexer.mll
+++ b/src/frontc/clexer.mll
@@ -184,7 +184,9 @@ let init_lexicon _ =
       ("__alignof", fun loc -> ALIGNOF loc);
       ("_Alignof", fun loc -> ALIGNOF loc);
       ("__alignof__", fun loc -> ALIGNOF loc);
+      ("alignof", fun loc -> ALIGNOF loc);
       ("_Alignas", fun loc -> ALIGNAS loc);
+      ("alignas", fun loc -> ALIGNAS loc);
       ("__volatile__", fun loc -> VOLATILE loc);
       ("__volatile", fun loc -> VOLATILE loc);
       ("__real__", fun loc -> REAL loc);

